snap/info.go

```diff
@@ -1005,9 +1006,20 @@ func (app *AppInfo) ServiceName() string {
 	return app.SecurityTag() + ".service"
 }
 
+func (app *AppInfo) serviceDir() string {
+	switch app.DaemonMode {
+	case SystemDaemon:
+		return dirs.SnapServicesDir
+	case UserDaemon:
+		return dirs.SnapUserServicesDir
```

Comment:
this path is not tested,  in general I don't see uses of UserDaemon in info_snap_yaml_test.go